File: week11/15/2_Tue_MinCostStairs.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();
        int mincost[n];
        for (int i = 0; i < n; i++){
            mincost[i] = 0;
        }
        mincost[1] = min(cost[0], cost[1]);
        for(int i=2; i<n; i++) {
            mincost[i] = min(mincost[i-2] + cost[i-1], mincost[i-1] + cost[i]);
        }
        return mincost[n-1];
    }
};

File: week11/15/2_Tue_MinCostStairs.py
from typing import List

class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        mincost = [0] * n 
        mincost[1] = min(cost[0], cost[1])
        for i in range(2, n):
            mincost[i] = min(mincost[i-1] + cost[i], mincost[i-2] + cost[i-1])
        return mincost[n-1]